### PR TITLE
Enable OpenId Tests

### PR DESCRIPTION
**Pull request by BrennanConroy**
@Tratcher

### CODE REVIEW

File: shared/Mocks/OpenIdConnect/OpenIdConnectBackChannelHttpHandler.cs [hunk 1]
```diff
@@ -24,6 +24,10 @@ protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage reques
             {
                 response.Content = new StringContent(File.ReadAllText(Path.Combine(basePath, "keys.json")));
             }
+            else if (request.RequestUri.AbsoluteUri == "https://login.windows.net/4afbc689-805b-48cf-a24c-d4aa3248a248/oauth2/token")
+            {
+                response.Content = new StringContent("{\"id_token\": \"id\", \"access_token\": \"access\"}");
```
Comment (BrennanConroy) [inline shared/Mocks/OpenIdConnect/OpenIdConnectBackChannelHttpHandler.cs]: Can we do something better here?

File: test/E2ETests/Implementation/OpenIdConnectLoginScenarios.cs [hunk 2]
```diff
@@ -96,17 +100,14 @@ public partial class Validator
...
             content = new FormUrlEncodedContent(formParameters.ToArray());
             // Need a non-redirecting handler
-            var handler = new HttpClientHandler() { AllowAutoRedirect = false };
+            var handler = new HttpClientHandler() { AllowAutoRedirect = true };
             handler.CookieContainer.Add(new Uri(_deploymentResult.ApplicationBaseUri), _httpClientHandler.CookieContainer.GetCookies(new Uri(_deploymentResult.ApplicationBaseUri)));
             _httpClient = new HttpClient(handler) { BaseAddress = new Uri(_deploymentResult.ApplicationBaseUri) };
 
             response = await DoPostAsync("Account/LogOff", content);
             Assert.Null(handler.CookieContainer.GetCookies(new Uri(_deploymentResult.ApplicationBaseUri)).GetCookieWithName(IdentityCookieName));
-            Assert.Equal<string>(
```
Comment (Tratcher) [inline test/E2ETests/Implementation/OpenIdConnectLoginScenarios.cs]: You're loosing some validation here by enabling auto-redirect.

**Comment by BrennanConroy** [comment]
:up: :date:

File: src/MusicStore/Controllers/AccountController.cs [hunk 3]
```diff
@@ -452,7 +452,7 @@ public ActionResult ExternalLogin(string provider, string returnUrl = null)
             var appEnv = HttpContext.RequestServices.GetService<IHostingEnvironment>();
             if (appEnv.EnvironmentName.StartsWith("OpenIdConnect"))
             {
-                await HttpContext.Authentication.SignOutAsync("OpenIdConnect");
+                return new SignOutResult("OpenIdConnect", new Microsoft.AspNetCore.Http.Authentication.AuthenticationProperties() { RedirectUri = Url.Action("Index", "Home") });
```
Comment (Tratcher) [inline src/MusicStore/Controllers/AccountController.cs]: remove namespace, wrap line.

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**